x-pack/plugins/elastic_assistant/server/routes/knowledge_base/get_knowledge_base_indices.ts

```diff
@@ -53,10 +53,12 @@ export const getKnowledgeBaseIndicesRoute = (router: ElasticAssistantPluginRoute
             include_unmapped: true,
           });
 
-          const indices = res.fields.content?.semantic_text?.indices;
-          if (indices) {
-            body.indices = Array.isArray(indices) ? indices : [indices];
+          const allIndices: string[] = [];
+          for (const [, value] of Object.entries(res.fields)) {
+            const indices = value.semantic_text?.indices ?? [];
+            allIndices.push(...(Array.isArray(indices) ? indices : [indices]));
           }
+          body.indices = [...new Set([...allIndices])].sort();
```

Comment:
nit to simplify a bit
```suggestion
          body.indices = Object.values(res.fields)
            .flatMap((value) => value.semantic_text?.indices ?? [])
            .filter((value, index, self) => self.indexOf(value) === index)
            .sort();
```